### DISCUSSION TITLE
Firmware Backup using wz_mini_hacks via SSH

### DISCUSSION THREAD

**Discussion by occultrefuse** [gtxaspec/wz_mini_hacks#802] [Q&A]
Hello,

I have a Cam v3 and a computer with macOS. Is it possible to create a backup of the original Wyze firmware using wz_mini_hacks via SSH? I'm considering installing _thingino_ on the camera long-term but I'd like to have a firmware backup in case I ever need to sell it or give it to someone else who isn't tech savvy. Thanks.

**Comment by HeyItsDaddy** [comment]
I'll put this thought out there for someone to validate...if you have any of the Wyze firmwares (most of us using multiple Cam V3's have 4.36.9.139 permanently on the root of a SD card used specifically for flashing firmware), and you put the camera into manual firmware flash mode (holding down Setup while plugging it in) would do the trick no matter what firmware you had on there at the time, right?  Unless there was some higher level root key/identifier it's also blowing away, I'd think this method would always work.

But, @gtxaspec would know all this and be able to answer in a heartbeat.  <3

**Reply by occultrefuse** [reply]
From the wiki at _themactep/thingino-firmware_ ([link](https://github.com/themactep/thingino-firmware/wiki/Wyze-%E2%80%90-Atom-%E2%80%90-Personal-(Hualai)-Devices)):

> If you don't perform a full backup of your camera's flash storage, installation on Wyze devices is a one-way process. Without a backup, you won't be able to restore Wyze services on the device, ever. The flash storage holds your unique information needed to connect to the Wyze cloud.

**Comment by endertable** [comment]
The big point of reference is backing up the .product_config file that is in the /configs directory.  That file contains your Wyze assigned MAC address, and your public key to be able to authenticate with Wyze services.  Without it, you will NOT be able to connect back with the Wyze ecosystem.

Then, upon restoration, you must be able to place this file back into the /configs directory, reboot, and reinitialize the camera with Wyze services.

**Reply by occultrefuse** [reply]
Just to add to your post, it seems that the `wz_mini_hacks` software creates a directory in the SD card root directory called `wyze_config_backup`. It contains a backup `configs` directory with everything from `[Wyze Camera Root]/configs`. It can be copied directly from the SD card to one's desktop.

**Comment by wes58** [comment]
If you are thinking about thingino you can use Ingenic USB Cloner Tool to backup your firmware - [https://thingino.com/cloner](url)
